### ISSUE TITLE
Making all fields of an object "nullable"

### ISSUE THREAD

**Issue by AslanTaheri** [colinhacks/zod#2893]
Currently, it is possible to make an object's properties optional by using `.Partial`. However, this does **NOT** make the object properties nullable, so there would be a repetitive task of adding `nullable()` to every field as seen below. Could we have an option to make all fields all nullable?

```
const MyObjectSchema = z.object({
   p1: z.string().nullable()
   p2: z.number().nullable()
   p3: z.boolean().nullable()
   p4: z.date().nullable()
   p5: z.symbol().nullable()
}).partial();
```

**Comment by cjlawson02** [comment]
+1, a nullish partial would be a great feature

**Comment by cicerotcv** [comment]
I agree and I would like to work on this feature.

**Comment by wesharper** [comment]
Has any progress been made here @cicerotcv? I'm happy to pick up where you left off.

**Comment by acro5piano** [comment]
An userland workaround is like this:

```typescript
export type NullableObjectReturn<T extends z.ZodRawShape> = z.ZodObject<
  { [k in keyof T]: z.ZodNullable<T[k]> },
  z.UnknownKeysParam,
  z.ZodTypeAny
>

export function nullableObject<T extends z.ZodRawShape>(
  obj: z.ZodObject<T>,
): NullableObjectReturn<T> {
  const newShape: any = {}

  keys(obj.shape).forEach((key) => {
    const fieldSchema = obj.shape[key]
    newShape[key] = fieldSchema.nullish()
  })

  return new z.ZodObject({
    ...(obj.shape._def as any),
    shape: () => newShape,
  }) as any
}
```

![image](https://github.com/user-attachments/assets/87ea09f0-d2ea-4c03-b00a-ca8a129eef89)


Hope it helps someone

**Comment by alex-aloia** [comment]
This would be extremely helpful as some of my APIs either return nulls or strip undefined keys out based on acceptType being json or avro. Also, having to add 'nullable' to every property is tedious.

**Comment by gdbjohnson** [comment]
Just to upvote, would be great to take an existing schema, and apply nullable() to it, like we can with partial().

**Comment by Povindu** [comment]
Is there any progress on this? I am also encountering this problem, and I also think a simple solution like ".partial()" would be good to have.

**Comment by cjlawson02** [comment]
Keep this open

**Comment by thewitcher-lab** [comment]
https://stackoverflow.com/questions/79835569/lost-openapi-descriptions-when-using-recursive-zod-schema-transformation-with-zo
Can someone check this question and help me find a solution? 
@cjlawson02  @colinhacks

**Comment by AslanTaheri** [comment]
> Hi, [@AslanTaheri](https://github.com/AslanTaheri). I'm [Dosu](https://dosu.dev), and I'm helping the Zod team manage their backlog and am marking this issue as stale.
> 
> **Issue Summary:**
> 
> * You requested a feature to make all fields of a Zod object schema nullable at once, similar to `.partial()` for optional fields.
> * The idea has strong community support, with multiple users expressing interest and willingness to contribute.
> * A userland workaround was shared, but no official implementation exists yet.
> * The issue has been kept open for discussion, with maintainers like [@colinhacks](https://github.com/colinhacks) asked to review the request.
> 
> **Next Steps:**
> 
> * Please let us know if this feature is still relevant for your use case with the latest version of Zod by commenting on this issue.
> * If we don’t hear back within 7 days, this issue will be automatically closed to keep the backlog manageable.
> 
> Thanks for your understanding and contribution!

Yes, this would be a great feature to have and still very much relevant when working with Zod. Thank you!

**Comment by LaDorade** [comment]
Still no news ? Seems fairly reasonable and straight-forward to do, no ?
